### DISCUSSION TITLE
Reaction fingerprints

### DISCUSSION THREAD

**Discussion by MartaPasquini** [rdkit/rdkit#5263] [Q&A] [answered]
Hello,

I'm working with fingerprints and similarity of reactions, with particular interest in understanding how the agents are handled and I run across an inconsistency.

When considering two reactions that only differ for the agent, such as 'CN.CC(O)=O>**O**>CNC(C)=O' and 'CN.CC(O)=O>**C**>CNC(C)=O', if the structural fingerprints are computed without additional arguments (rdChemReactions.CreateStructuralFingerprintForReaction()) the Tanimoto similarity is 0.979. However, if ReactionFingerprintParams() is used while creating the fingerprints (without changing the default values) the returned similarity has a different value (1.0). 

![ReactionFingerprints](https://user-images.githubusercontent.com/99722467/166882090-686c72db-2f7d-4e9f-995e-836f9a814053.png)

Could it be related to a different set of default values triggered when the optional ReactionFingerprintParams() is included? The default parameters for ReactionFingerPrintParams are reported here https://www.rdkit.org/docs/cppapi/structRDKit_1_1ReactionFingerprintParams.html, but which are those used when ReactionFingerprintParams() is not used?

Thank you!

**Comment by greglandrum** [comment, marked as answer]
Hi @MartaPasquini, this is unfortunately not documented anywhere and you have to look in the C++ code.

If you call `CreateStructuralFingerprintForReaction()` without arguments, it uses `DefaultStructuralFPParams`, which is an instance of `ReactionFingerprintParams` with some of the values changed.

The values are defined here:
https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/ChemReactions/ReactionFingerprints.cpp#L123

Here's the declaration of the `ReactionFingerprintParams` class, where you can see the ordering of the arguments to the constructor as well as the default values (which are not used by `DefaultStructuralFPParams`nm):
https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/ChemReactions/ReactionFingerprints.h#L78
```
                            bool includeAgents, double bitRatioAgents,
                            unsigned int nonAgentWeight, int agentWeight,
                            unsigned int fpSize, FingerprintType fpType
```

Sorry this amount of detective work is required...
